server/api/api.go

```diff
@@ -2087,7 +2108,17 @@ func (a *API) handleGetTeamUsers(w http.ResponseWriter, r *http.Request) {
 	auditRec := a.makeAuditRecord(r, "getUsers", audit.Fail)
 	defer a.audit.LogRecord(audit.LevelRead, auditRec)
 
-	users, err := a.app.SearchTeamUsers(teamID, searchQuery)
+	isGuest, err := a.userIsGuest(userID)
```

Comment:
Is it possible to move some of this logic to the permissions service, and treat being a guest more like a set of permissions?  Otherwise every rest API gets this same code pattern checking if guest then two "if" statements checking for error and then doing something if guest.